samza-core/src/main/java/org/apache/samza/config/StorageConfig.java

```diff
@@ -225,10 +226,13 @@ public boolean hasDurableStores() {
   }
 
   /**
-   * Helper method to get the number of stores configured with a changelog.
+   * Helper method to get the number of persistent stores.
    */
-  public int getNumStoresWithChangelog() {
-    Config subConfig = subset(STORE_PREFIX, true);
-    return new Long(subConfig.keySet().stream().filter(key -> key.endsWith(CHANGELOG_SUFFIX)).count()).intValue();
+  public int getNumPersistentStores() {
+    return (int) getStoreNames().stream()
+        .map(storeName -> getStorageFactoryClassName(storeName))
+        .filter(factoryName -> factoryName.isPresent())
+        .filter(factoryName -> factoryName.get().equals(PERSISTENT_STORE_FACTORY))
```

Comment:
This is inconsistent with `StoreProperties.isPersistedToDisk`. Over there, it is "persisted" if it is not an in-memory store. Should we do the same here?

Reply:
Ah, yes, I was looking for exactly that. 
Had to add a constant because of the module dependency.